service-base/src/main/java/org/eclipse/hono/service/http/TracingHandler.java

```diff
@@ -0,0 +1,163 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.service.http;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.eclipse.hono.tracing.MultiMapExtractAdapter;
+
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+import io.opentracing.propagation.Format;
+import io.opentracing.tag.Tags;
+import io.vertx.core.Handler;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * Handler which creates tracing data for all server requests. It should be added to
+ * {@link io.vertx.ext.web.Route#handler(Handler)} and {@link io.vertx.ext.web.Route#failureHandler(Handler)} as the
+ * first in the chain.
+ * <p>
+ * This class has been copied from the <a href="https://github.com/opentracing-contrib/java-vertx-web">
```

Comment:
I do think that we still need a CQ upfront when we take over code into our repository.

Reply:
Yes, we already have a CQ for this: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=16576